boot/boot.js

```diff
@@ -1444,6 +1444,7 @@ $tw.loadTiddlersBrowser = function() {
 	for(var t=0; t<containerIds.length; t++) {
 		$tw.wiki.addTiddlers($tw.wiki.deserializeTiddlers("(DOM)",document.getElementById(containerIds[t])));
 	}
+	callback();
```

Comment:
@Arlen22 
> loadTiddlersNode is also synchronous, so it would apply the same way. I had to do both synchronous or neither, it seemed, but I technically would not have to. 

Any code can be made asynchronous with the proper handling. Even if the 99% of the function code is synchronous. 

> It seems to me that that only applies as a matter of principle. 
 
Are you familiar with the event loop ? It's a sincere question, I want to know if you know about it or not. But for your information, this is not just a matter of principle. Using callbacks for both synchronously and asynchronously is one of the worst javascript bad practices. Synchronous functions have a very good and specific way of returning the control to the main process, which is the return statement. Callbacks are used when returning on the main thread is not possible. if the function is synchronous, then you just have to call it and at the very next line the code will be completed. No need for a callback.

> It does not matter here that we are sync'ly calling the callback, because the callback code is in boot.js and therefore is under our control anyway

I don't understand this sentence. You are in control of boot.js (at this moment) but not of the code that is passing you a callback, expecting it to be called on an asynchronous way.

The main question is, why are you providing an asynchronous interface for a synchronous flow?

I'm not saying that the boot process should not be improved to allow an asynchronous flow, but I think it should be totally, not just look like it.

> however the way I have it would allow someone to overload the loadTiddlersBrowser as well if they wanted to. 

Again I don't understand this sentence. Could you expand on it?